optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConnectionFollowRedirects.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// credits for https://www.cs.mun.ca/java-api-1.5/guide/deployment/deployment-guide/upgrade-guide/article-17.html
+
+package org.optaplanner.examples.conferencescheduling.persistence;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+
+public class ConnectionFollowRedirects {
```

Comment:
Let's talk about this redirects implementation during our call.
I agree we need a way to mock out the rest service.
But I am not keen on adding all this complexity and especially the dependency into dep tree.
I hope we can simply supply a "file://" protocol as the baseUrl to mock it out magically.